src/plugin/memory-provider.ts

```diff
@@ -14,34 +14,43 @@
  * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
  ********************************************************************************/
 
-import * as vscode from 'vscode';
-import * as manifest from './manifest';
 import { DebugProtocol } from '@vscode/debugprotocol';
-import { MemoryReadResult, MemoryWriteResult } from '../common/messaging';
+import * as vscode from 'vscode';
+import { VariableRange, WrittenMemory } from '../common/memory-range';
+import { ReadMemoryResult, SessionContext, WriteMemoryResult } from '../common/messaging';
 import { AdapterRegistry } from './adapter-registry/adapter-registry';
-import { VariableRange } from '../common/memory-range';
+import * as manifest from './manifest';
+import { isDebugEvent, isDebugRequest, isDebugResponse, sendRequest } from '../common/debug-requests';
+import { stringToBytesMemory } from '../common/memory';
 
 export interface LabeledUint8Array extends Uint8Array {
     label?: string;
 }
 
-// eslint-disable-next-line @typescript-eslint/no-explicit-any
-const isInitializeMessage = (message: any): message is DebugProtocol.InitializeResponse => message.command === 'initialize' && message.type === 'response';
-// eslint-disable-next-line @typescript-eslint/no-explicit-any
-const isStoppedEvent = (message: any): boolean => message.type === 'event' && message.event === 'stopped';
-
 export class MemoryProvider {
     public static ReadKey = `${manifest.PACKAGE_NAME}.canRead`;
     public static WriteKey = `${manifest.PACKAGE_NAME}.canWrite`;
 
-    private _onDidStopDebug: vscode.EventEmitter<vscode.DebugSession> = new vscode.EventEmitter<vscode.DebugSession>();
-    public readonly onDidStopDebug: vscode.Event<vscode.DebugSession> = this._onDidStopDebug.event;
+    private _onDidStopDebug = new vscode.EventEmitter<vscode.DebugSession>();
+    public readonly onDidStopDebug = this._onDidStopDebug.event;
+
+    private _onDidWriteMemory = new vscode.EventEmitter<WrittenMemory>();
+    public readonly onDidWriteMemory = this._onDidWriteMemory.event;
 
-    protected readonly sessions = new Map<string, DebugProtocol.Capabilities | undefined>();
+    private _sessionContext: SessionContext = { canRead: false, canWrite: false };
```

Comment:
Good catch! I had that in there because my context also supports sending the session id and that id is not part of the `capabilities` response. However, I only had that in for logging and otherwise the session id is not really used so I'll remove it for now. Probably we'll get something similar when we work on #97.

Reply:
Good catch! I'll remove it.